queenbee/schema/function.py

```diff
@@ -134,24 +158,24 @@ def validate_variable(variables, func_name, input_names):
                 raise ValueError(
                     '{{%s}} in %s function does not follow x.y.z pattern'% (m, func_name)
                 )
-            else:
-                if not ref in ('inputs', 'outputs', 'workflow'):
-                    raise ValueError(
-                        'Referenced values must start with inputs, outputs or workflow'
-                    )
-                if ref == 'workflow':
-                    warnings.warn(
-                        '[{}]: Referencing workflow parameters in a function '
-                        ' makes the function less reusable. Use inputs of the function'
-                        ' instead and assign workflow values in flow section when calling'
-                        ' this function. Reference: {}'.format(func_name, m)
-                    )
-                    continue
+
+            if not ref in ('inputs', 'outputs', 'workflow'):
+                raise ValueError(
+                    'Referenced values must start with inputs, outputs or workflow'
+                )
+            if ref == 'workflow':
+                warnings.warn(
+                    '[{}]: Referencing workflow parameters in a function '
+                    ' makes the function less reusable. Use inputs of the function'
+                    ' instead and assign workflow values in flow section when calling'
+                    ' this function. Reference: {}'.format(func_name, m)
+                )
+                continue
             
             # now ensure input references are valid
             if ref != 'inputs' or typ != 'parameters':
                 continue
 
             assert name in input_names, \
-                'Illegal input parameter name in "%s": {{%s}}\n' \
+                'Illegal output parameter name in "%s": {{%s}}\n' \
```

Comment:
why output and should we change the next line to output too?

Reply:
Oh that's my bad. I think the check run by this function applies to both inputs and outputs. Think we can leave it for now and raise an issue to generally improve messaging/debugging.